File: job-hunt-frontend/job-hunt-ui/src/services/TitleService.js
import axios from "axios";
import { getToken } from "./AuthService";

const TITLE_REST_API_BASE_URL = 'http://localhost:8080/api/titles'

// Add a request interceptor
axios.interceptors.request.use(function (config) {
    config.headers['Authorization'] = getToken();
    return config;
  }, function (error) {
    return Promise.reject(error);
  });

export const getAllTitles = () => axios.get(TITLE_REST_API_BASE_URL); // GET ALL

export const addTitle = (title) => axios.post(TITLE_REST_API_BASE_URL, title); // ADD

export const deleteTitle = (id) => axios.delete(TITLE_REST_API_BASE_URL + '/' + id); // DELETE

export const getTitle = (id) => axios.get(TITLE_REST_API_BASE_URL + '/' + id); // GET

export const updateTitle = (id, title) => axios.put(TITLE_REST_API_BASE_URL + '/' + id, title); // UPDATE

File: job-hunt/src/main/java/ykpsph/jobhunt/service/TitleService.java
package ykpsph.jobhunt.service;

import ykpsph.jobhunt.dto.TitleDTO;

import java.util.List;

public interface TitleService {
    TitleDTO addTitle(TitleDTO titleDTO); // ADD
    List<TitleDTO> getAllTitles(); // GET ALL
    TitleDTO getTitle(Long id); // GET

    TitleDTO updateTitle(TitleDTO titleDTO, Long id); // UPDATE

    void deleteTitle(Long id); // DELETE
}
